option.go

```diff
@@ -305,16 +314,12 @@ func (optSet *OptionSet) SetDefaults() error {
 
 	var merr *multierror.Error
 
-	for i, opt := range *optSet {
-		// Skip values that may have already been set by the user.
-		if opt.ValueSource != ValueSourceNone {
-			continue
-		}
```

Comment:
Review: This was removed in this refactor because we have a different way of handling values that have been set by the user.

However, this breaks _one_ test in coder/coder: https://github.com/coder/coder/actions/runs/11782339194/job/32817210417?pr=15476#step:7:621

I could add a `if opt.Value == nil && opt.ValueSource != ValueSourceNone { continue }` here and it will fix the test. The only problem is, I don't understand the logic. If the value has been provided by the user and `opt.Value == nil`, that's OK. But if the value hasn't been provided, then we error out. Why?

Reply:
I haven't spent time to try and figure this out. This felt vaguely familiar, so I looked up when it was added.

It was added in this commit: https://github.com/coder/coder/commit/7edea99f8a679761fabbca4abe280370f8ea10f7

And this PR: https://github.com/coder/coder/pull/6934

So it has to relate with YAML support somehow? I was hoping the commit when it was added would have more context.

Reply:
Thanks for looking and sharing, that was very helpful! It lead me to this: [`e053fda` (#23)](https://github.com/coder/serpent/pull/23/commits/e053fdac0b0dd01a0e6278dfbd703bbf7486a840).